pkg/cluster/cluster.go

```diff
@@ -232,15 +237,62 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				c.status.Control()
 			}
 
-			// Just here get the secret version in case there is one.
 			if c.cluster.Spec.Auth != nil {
-				// Look for updates in the secret used for auth then
-				// trigger config reload in case there are new updates.
-				authSecret := c.cluster.Spec.Auth.ClientsAuthSecret
-				result, err := c.config.KubeCli.Secrets(c.cluster.Namespace).Get(authSecret, metav1.GetOptions{})
-				if err == nil && secretLastResourceVersion != result.ResourceVersion {
-					secretLastResourceVersion = result.ResourceVersion
-					c.updateConfigMap()
+				if c.cluster.Spec.Auth.ClientsAuthSecret != "" {
```

Comment:
Just a suggestion - you could move this into a checkCliientAuthUpdate function for readability...